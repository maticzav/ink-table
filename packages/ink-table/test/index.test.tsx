import React from "react";
import { Text } from "ink";
import stripAnsi from "strip-ansi";
import { test, expect } from "vitest";
import { render } from "ink-testing-library";

import { Table } from "../src/index.js";

const Custom = ({ children }: React.PropsWithChildren<{}>) => (
  <Text italic color="red">
    {children}
  </Text>
);

// Tests ---------------------------------------------------------------------

test("Renders table.", () => {
  const data = [{ name: "Foo" }];

  const { lastFrame: actual } = render(<Table data={data} />);

  expect(stripAnsi("\n" + actual())).toMatchSnapshot();
});

test("Renders table with numbers.", () => {
  const data = [{ age: 12, name: "Foo" }];
  const { lastFrame: actual } = render(<Table data={data} />);

  expect(stripAnsi("\n" + actual())).toMatchSnapshot();
});

test("Renders table with multiple rows.", () => {
  const data = [
    { age: 12, name: "Foo" },
    { age: 0, name: "Bar" },
  ];
  const { lastFrame: actual } = render(<Table data={data} />);

  expect(stripAnsi("\n" + actual())).toMatchSnapshot();
});

test("Renders table with undefined value.", () => {
  const data = [{ name: "Foo" }, { age: 15, name: "Bar" }];
  const { lastFrame: actual } = render(<Table data={data} />);

  expect(stripAnsi("\n" + actual())).toMatchSnapshot();
});

test("Renders table with custom padding.", () => {
  const data = [
    { age: 12, name: "Foo" },
    { age: 15, name: "Bar" },
  ];
  const { lastFrame: actual } = render(<Table data={data} padding={3} />);

  expect(stripAnsi("\n" + actual())).toMatchSnapshot();
});

test("Renders table with custom header.", () => {
  const data = [
    { age: 12, name: "Foo" },
    { age: 15, name: "Bar" },
  ];
  const { lastFrame: actual } = render(<Table data={data} header={Custom} />);

  expect(stripAnsi("\n" + actual())).toMatchSnapshot();
});

test("Renders table with custom cell.", () => {
  const data = [
    { age: 12, name: "Foo" },
    { age: 15, name: "Bar" },
  ];
  const { lastFrame: actual } = render(<Table data={data} cell={Custom} />);

  expect(stripAnsi("\n" + actual())).toMatchSnapshot();
});

test("Renders table with custom skeleton.", () => {
  const data = [
    { age: 12, name: "Foo" },
    { age: 15, name: "Bar" },
  ];
  const { lastFrame: actual } = render(<Table data={data} skeleton={Custom} />);

  expect(stripAnsi("\n" + actual())).toMatchSnapshot();
});

// ---------------------------------------------------------------------------
