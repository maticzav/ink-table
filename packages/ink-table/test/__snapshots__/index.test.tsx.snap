// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Renders table with custom cell. 1`] = `
"
┌─────┬──────┐
│ age │ name │
├─────┼──────┤
│ 12  │ Foo  │
├─────┼──────┤
│ 15  │ Bar  │
└─────┴──────┘"
`;

exports[`Renders table with custom header. 1`] = `
"
┌─────┬──────┐
│ age │ name │
├─────┼──────┤
│ 12  │ Foo  │
├─────┼──────┤
│ 15  │ Bar  │
└─────┴──────┘"
`;

exports[`Renders table with custom padding. 1`] = `
"
┌─────────┬──────────┐
│   age   │   name   │
├─────────┼──────────┤
│   12    │   Foo    │
├─────────┼──────────┤
│   15    │   Bar    │
└─────────┴──────────┘"
`;

exports[`Renders table with custom skeleton. 1`] = `
"
┌─────┬──────┐
│ age │ name │
├─────┼──────┤
│ 12  │ Foo  │
├─────┼──────┤
│ 15  │ Bar  │
└─────┴──────┘"
`;

exports[`Renders table with multiple rows. 1`] = `
"
┌─────┬──────┐
│ age │ name │
├─────┼──────┤
│ 12  │ Foo  │
├─────┼──────┤
│ 0   │ Bar  │
└─────┴──────┘"
`;

exports[`Renders table with numbers. 1`] = `
"
┌─────┬──────┐
│ age │ name │
├─────┼──────┤
│ 12  │ Foo  │
└─────┴──────┘"
`;

exports[`Renders table with undefined value. 1`] = `
"
┌──────┬─────┐
│ name │ age │
├──────┼─────┤
│ Foo  │     │
├──────┼─────┤
│ Bar  │ 15  │
└──────┴─────┘"
`;

exports[`Renders table. 1`] = `
"
┌──────┐
│ name │
├──────┤
│ Foo  │
└──────┘"
`;
